source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+ * you use to define the region of your ellipsoid chunk geometry.
+ *
+ * This plugin is developed by Tahiry Rajaonarison, D. Sarah Stamps,
+ * and Wolfgang Bangerth.
+ */
+
+
+namespace aspect
+{
+  namespace InitialConditions
+  {
+    using namespace dealii;
+
+    /**
+     * Function that returns latitude and longitude from ECEF Cartesian
+     * coordinates that account for ellipsoidal shape of the Earth
+     * with WGS84 parameters.
+     */
+
+    template <int dim>
+    std::pair<double, double>
+    AdiabaticBoundary<dim>::lat_long_from_xyz_WGS84(const Point<3> &pos) const
+    {
+      /* Define WGS84 ellipsoid constants. */
+      const double radius = 6378137;
+      const double ellipticity = 8.1819190842622e-2;
+
+      const double b = std::sqrt(radius * radius
+                                 * (1 - ellipticity * ellipticity));
+      const double ep = std::sqrt((radius * radius - b * b) / (b * b));
+      const double p = std::sqrt(pos(0) * pos(0) + pos(1) * pos(1));
+      const double th = std::atan2(radius * pos(2), b * p);
+      const double lon = std::atan2(pos(1), pos(0));
+      const double lat = std::atan2((pos(2) + ep * ep * b * std::sin(th)
+                                     * std::sin(th) * std::sin(th)),
+                                    (p - (ellipticity
+                                          * ellipticity
+                                          * radius
+                                          * (std::cos(th) * std::cos(th)
+                                             * std::cos(th)))));
+
+      /* Convert to degrees. */
+      const double lon_degrees = lon * (180 / numbers::PI);
+      const double lat_degrees = lat * (180 / numbers::PI);
+
+      /* Set all longitudes between [0,360]. */
+      if (lon_degrees < 0)
+        return std::make_pair(lat_degrees, lon_degrees + 360);
+      else if (lon_degrees > 360)
+        return std::make_pair(lat_degrees, lon_degrees - 360);
+      else
+        return std::make_pair(lat_degrees, lon_degrees);
+
+    }
+
+    template <>
+    double
+    AdiabaticBoundary<2>::get_isotherm_depth(const double,
+                                             const double) const
+    {
+      AssertThrow(false, ExcInternalError());
+      return 0;
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::get_isotherm_depth(const double latitude,
+                                               const double longitude) const
+    {
+      /**
+       * Loop over the entire array and see if we find a point
+       * that's within delta of what we're looking for. If this
+       * point is not within the delta then test if it is within twice
+       * as much as delta.
+       */
+      for (unsigned int i = 0; i <= depths_iso.size();)
+        if (std::fabs(latitude - latitudes_iso[i]) <= delta && std::fabs(longitude - longitudes_iso[i]) < delta)
+          {
+            return -depths_iso[i]*1000;
```

Comment:
Does this 1000 come from the conversion from km to m? Could you then move it into the place where you read in the data and document its origin there? That way this function becomes easier to understand. Also where does the minus come from?